### PR TITLE
Refactor plus minimal star map hud

### PR DESCRIPTION
**Pull request by SorraTheOrc**
This is mostly a refactoring to start to move data into a central GameManager Singleton. The intent is for GameManager to have all the top level configuration and for it to cascade down from there. This was required to implement the StarMap HUD as we had a circular reference between Galaxy -> Solar System -> Planet. Untangling this was complex, so I have attempted to simplify.

For the StarMap HUD we just have a count of the number of populate vs unpopulated planets right now.

### CODE REVIEW

**Comment by SorraTheOrc** [comment]
Don't have factions present in every solar system apply them at random and provide a summary in the Sta
rMap HUD. This provides a framework for having factions that own areas of the galaxy, but right now it is entirely rando
m.

File: Assets/_Project/Scripts/GameManager.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        int seed = 0;
+        [SerializeField, Tooltip("The Galaxy Config is at the root of the definition of a galaxy and everything within it." +
+            "It is essential to generating data within a glaxy.")]
+        internal GalaxyConfig galaxyConfig;
+        [SerializeField, Tooltip("The Solar System Instance used to generate the solar systems within the game.")]
+        internal SolarSystemInstance solarSystemInstance;
+
+        public static GameManager Instance { get; private set; }
+
+        private void Awake()
```
Comment (yolanother) [inline Assets/_Project/Scripts/GameManager.cs]: Should this be don't destroy on load?

Comment (SorraTheOrc) [reply Assets/_Project/Scripts/GameManager.cs]: I don't know, educate me on the advantages/disadvantages.

Comment (SorraTheOrc) [reply Assets/_Project/Scripts/GameManager.cs]: As per convo on discord I made it don't destroy on load as a singleton.

File: Assets/_Project/Scripts/GameManager.cs [hunk 2]
```diff
@@ -28,6 +28,7 @@ public class GameManager : MonoBehaviour
         private void Awake()
         {
             Instance = this;
+            DontDestroyOnLoad(Instance);
```
Comment (yolanother) [inline Assets/_Project/Scripts/GameManager.cs]: You need to handle destroy for other scenes that have the game manager.

https://gist.github.com/yolanother/3ae865b6a80f2f45940f18645c308f20

Comment (SorraTheOrc) [reply Assets/_Project/Scripts/GameManager.cs]: done

**Pull request merged**